File: solutions/c/kindergarten-garden/1/kindergarten_garden.c
#include <string.h>
#include "kindergarten_garden.h"

static plant_t plant(char c)
{
  switch(c) {
    default:
    case 'C':
      return CLOVER;
    case 'G':
      return GRASS;
    case 'R':
      return RADISHES;
    case 'V':
      return VIOLETS;
  }
}

plants_t plants(const char* diagram, const char* student)
{
  const int index = student[0] - 'A';
  const int row_1_offset = 0;
  const int row_2_offset = strlen(diagram) / 2 + 1;

  return (plants_t){
    .plants = {
      plant(diagram[row_1_offset + index * 2]),
      plant(diagram[row_1_offset + index * 2 + 1]),
      plant(diagram[row_2_offset + index * 2]),
      plant(diagram[row_2_offset + index * 2 + 1]),
    }
  };
}

File: solutions/javascript/kindergarten-garden/1/kindergarten-garden.js
//
// This is only a SKELETON file for the 'Kindergarten Garden' exercise.
// It's been provided as a convenience to get you started writing code faster.
//

const DEFAULT_STUDENTS = ['Alice','Bob','Charlie','David','Eve','Fred','Ginny','Harriet','Ileana','Joseph','Kincaid','Larry',];

const PLANT_CODES = {
  G: 'grass',
  V: 'violets',
  R: 'radishes',
  C: 'clover',
};

export class Garden {
  constructor(diagram, students = DEFAULT_STUDENTS) {
    this.diagram = diagram.split('\n')
    this.students = students.sort()
  }

  plants(student) {
    let i = this.students.indexOf(student)
    return this.diagram.map(row => row.slice(2 * i, 2 * i + 2)).flatMap(row => [...row].map(p => PLANT_CODES[p]))
  }
}
